sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -195,67 +195,133 @@ case class DropFunction(
     isTemp: Boolean)(sql: String)
   extends NativeDDLCommand(sql) with Logging
 
-/** Rename in ALTER TABLE/VIEW: change the name of a table/view to a different name. */
+/**
+ * A command that renames a table/view.
+ *
+ * The syntax of this command is:
+ * {{{
+ *    ALTER TABLE table1 RENAME TO table2;
+ *    ALTER VIEW view1 RENAME TO view2;
+ * }}}
+ */
 case class AlterTableRename(
     oldName: TableIdentifier,
-    newName: TableIdentifier)(sql: String)
-  extends NativeDDLCommand(sql) with Logging
+    newName: TableIdentifier)
+  extends RunnableCommand {
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    sqlContext.sessionState.catalog.renameTable(oldName, newName)
+    Seq.empty[Row]
```

Comment:
Here, based on my understanding, two extra steps are needed for table rename. 

First, in `HiveSessionCatalog`, the function `renameTable` needs to invalidate the Hive Megastore Cache by `hiveContext.invalidateTable(tableName)`. 

Second, here, we need an efficient way to refresh `CacheManager`. 

Please correct me if I am wrong. Thanks!